cmd/minishift/cmd/start_preflight_windows.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/minishift/minishift/pkg/util/filehelper"
+
+	"github.com/golang/glog"
+	"golang.org/x/sys/windows/registry"
+)
+
+// checkVBoxInstalled checks for the locations of VBoxManage from environmet variables,
+// in the defaiult install location and in windows registry respectivley, returns true
+// if found in any one of the location, https://github.com/docker/machine/blob/master/drivers/virtualbox/virtualbox_windows.go
+func checkVBoxInstalled() bool {
+	vboxCmd := "VBoxManage.exe"
+
+	if p := os.Getenv("VBOX_INSTALL_PATH"); p != "" {
+		if filehelper.Exists(filepath.Join(p, vboxCmd)) {
+			return true
+		}
+	}
+
+	if p := os.Getenv("VBOX_MSI_INSTALL_PATH"); p != "" {
+		if filehelper.Exists(filepath.Join(p, vboxCmd)) {
+			return true
+		}
+	}
+	// Look in default installation path for VirtualBox version > 5
+	if filehelper.Exists(filepath.Join("C:", "Program Files", "Oracle", "VirtualBox", vboxCmd)) {
+		return true
+	}
+	// Look in windows registry
+	if p, err := findVBoxInstallDirInRegistry(); err != nil {
+		if filehelper.Exists(filepath.Join(p, vboxCmd)) {
+			return true
+		}
+	}
+	return false
```

Comment:
What happened to the fallback? This is not at all the same as in virtualbox_windows.go. There the methods ends with - `return detectVBoxManageCmdInPath() //fallback to path`. This is not here at all, making your code too stringent again.

Reply:
Yes @hferentschik the fallback to checking in path is not present here. Will add it.